### PR TITLE
git-sync: darwin support

### PR DESCRIPTION
**Pull request by ryane**
### Description

`services.git-sync` support on darwin

- On darwin, creates a launch agent to run git-sync on an interval and when the `path` changes.
- The `uri` option is not used on Darwin. The auto-creation of the local git directory from the `uri` is a feature of the
[git-sync-on-inotify](https://github.com/simonthum/git-sync/blob/master/contrib/git-sync-on-inotify) wrapper (which won't work on Darwin afaik) and not `git-sync` itself.

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [x] Code tested through `nix-shell --pure tests -A run.all` or `nix develop --ignore-environment .#all` using Flakes.

- [x] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/docs/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [x] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

#### Maintainer CC

@imalison  @cab404

_Disclaimer: new-ish to nix, first attempted home-manager contribution_

### CODE REVIEW

**Comment by cab404** [comment]
Can't test, but LGTM!

File: modules/services/git-sync.nix [hunk 1]
```diff
@@ -90,12 +103,14 @@ in {
...
-  config = mkIf cfg.enable {
-    assertions = [
-      (lib.hm.assertions.assertPlatform "services.git-sync" pkgs
-        lib.platforms.linux)
-    ];
+  config = mkIf cfg.enable (mkMerge [
+    (mkIf pkgs.stdenv.isLinux { systemd.user.services = services; })
+    (mkIf pkgs.stdenv.isDarwin {
+      launchd = {
+        enable = true;
```
Comment (rycee) [inline modules/services/git-sync.nix:110]: Don't think this is necessary, it should default to true on Darwin.

File: modules/lib/maintainers.nix [hunk 2]
```diff
@@ -171,6 +171,14 @@
...
     githubId = 1224006;
   };
+  ryane = {
+    email = "ryanesc@gmail.com";
+    github = "ryane";
+    githubId = 7346;
+    name = "Ryan Eschinger";
+    keys =
+      [{ fingerprint = "E4F4 1EAB BF0F C785 06D8  62EF EF68 CF41 D42A 593D"; }];
+  };
```
Comment (rycee) [inline modules/lib/maintainers.nix:181]: No need to add yourself if you are already in Nixpkgs' maintainer list.

**Comment by rycee** [comment]
Thanks! Merged now 🙂

**Comment by rycee** [comment]
@ambroisie Fair enough, added now 🙂

**Pull request merged**